lib/screens/settings.dart

```diff
@@ -0,0 +1,134 @@
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../providers/theme_provider.dart';
+
+
+
+
+class Settings extends StatefulWidget {
+  static String routename = 'SettingsPage';
+  const Settings({Key? key}) : super(key: key);
+
+  @override
+  State<Settings> createState() => _SettingsState();
+}
+
+class _SettingsState extends State<Settings> {
+
+
+
+  Map<String, int> themes = {"System Default":1, "Light":2, "Dark":3};
+
+
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Theme.of(context).primaryColor,
+        elevation: 5.0,
+        //shadowColor: Colors.black26,
+        shape: const BeveledRectangleBorder(
+          borderRadius: BorderRadius.all(Radius.circular(10)),
+        ),
+        title: Text(
+          'Settings',
+          style: TextStyle(
+            color: Theme.of(context).secondaryHeaderColor,
+          ),
+        ),
+      ),
+      body: SafeArea(
+        child: ListView(
+          children: [
+            ListTile(
+              title: const Text(
+                "Theme",
+              ),
+              onTap: () {
+                showDialog(
+                  context: context,
+                  builder: (dialogContext) {
+                    return StatefulBuilder(builder: (stfContext, stfSetState) {
+                      return AlertDialog(
+                        title: const Padding(
+                          padding: EdgeInsets.all(8.0),
+                          child: Center(
+                            child: Text(
+                              "Set Theme",
+                            ),
+                          ),
+                        ),
+                        content: SizedBox(
+                          height: 160,
+                          width: 150,
+                          child: ListView(
+                            children: themes.keys.map((e) => RadioListTile(
+                              title: Text(e),
+                              value: themes[e],
+                              groupValue: themes[device.theme],
+                              activeColor: Colors.cyan,
+                              onChanged: (value) {
+                                stfSetState(() {
+                                  device.theme = e;
+                                });
+                              },
+                            )).toList(),
+                          ),
+                        ),
+                        actions: [
+                          TextButton(onPressed: () {
+                               Navigator.of(context).pop();
+                               Provider.of<ThemeProvider>(context, listen: false).changeTheme(device.theme);
```

Comment:
![image](https://github.com/IEEE-VIT/hacktoberfest-flutter/assets/84137299/11e4f02a-72be-473a-96e1-ae92df8ef15a)

Please check if this error exist or its in my editor

Reply:
Hey @dk-a-dev, I have updated the changeTheme function in the ThemeProvider screen, which requires a positional argument. This is the reason you are encountering an error.
![image](https://github.com/IEEE-VIT/hacktoberfest-flutter/assets/129930078/8ea70968-0489-4495-9737-144c35d96fef)
![image](https://github.com/IEEE-VIT/hacktoberfest-flutter/assets/129930078/60f11d0c-b8cc-461c-9593-ce0acde084fa)